tests/byzantium/eip198_modexp_precompile/test_modexp.py

```diff
@@ -0,0 +1,273 @@
+            + self.base
+            + self.exponent
+            + self.modulus
+            + self.extra_data
+        )
+
+
+@dataclass(kw_only=True, frozen=True, repr=False)
+class ModExpRawInput(TestParameterGroup):
+    """
+    Helper class to directly define a raw input to the MODEXP precompile.
+    """
+
+    raw_input: str
+
+    def create_modexp_tx_data(self):
+        """
+        The raw input is already the MODEXP precompile input.
+        """
+        return self.raw_input
+
+
+@dataclass(kw_only=True, frozen=True, repr=False)
+class ExpectedOutput(TestParameterGroup):
+    """
+    Expected test result.
+
+    Attributes:
+        call_return_code (str): The return_code from CALL, 0 indicates unsuccessful call
+            (out-of-gas), 1 indicates call succeeded.
+        returned_data (str): The output returnData is the expected output of the call
+    """
+
+    call_return_code: str
+    returned_data: str
+
+
+@pytest.mark.valid_from("Byzantium")
+@pytest.mark.parametrize(
+    ["input", "output"],
+    [
+        (
+            ModExpInput(base="", exponent="", modulus="02"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x01"),
+        ),
+        (
+            ModExpInput(base="", exponent="", modulus="0002"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x0001"),
+        ),
+        (
+            ModExpInput(base="00", exponent="00", modulus="02"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x01"),
+        ),
+        (
+            ModExpInput(base="", exponent="01", modulus="02"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x00"),
+        ),
+        (
+            ModExpInput(base="01", exponent="01", modulus="02"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x01"),
+        ),
+        (
+            ModExpInput(base="02", exponent="01", modulus="03"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x02"),
+        ),
+        (
+            ModExpInput(base="02", exponent="02", modulus="05"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x04"),
+        ),
+        (
+            ModExpInput(base="", exponent="", modulus=""),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x"),
+        ),
+        (
+            ModExpInput(base="", exponent="", modulus="00"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x00"),
+        ),
+        (
+            ModExpInput(base="", exponent="", modulus="01"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x00"),
+        ),
+        (
+            ModExpInput(base="", exponent="", modulus="0001"),
+            ExpectedOutput(call_return_code="0x01", returned_data="0x0000"),
+        ),
+        # Test cases from EIP 198 (Note: the cases where the call goes out-of-gas and the
+        # final test case are not yet tested)
```

Comment:
I all the test cases from the EIP have been implemented now?
```suggestion
        # Test cases from EIP 198.
```

Reply:
Yup :smile: